backend/LexBoxApi/GraphQL/OrgMutations.cs

```diff
@@ -0,0 +1,88 @@
+﻿using LexBoxApi.Auth;
+using LexCore.Entities;
+using LexCore.Exceptions;
+using LexCore.ServiceInterfaces;
+using LexData;
+using LexData.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace LexBoxApi.GraphQL;
+
+[MutationType]
+public class OrgMutations
+{
+    [Error<DbError>]
+    [UseMutationConvention]
+    [UseFirstOrDefault]
+    [UseProjection]
+    public async Task<IQueryable<Organization>> CreateOrganization(string name,
+        LexBoxDbContext dbContext,
+        LoggedInContext loggedInContext,
+        IPermissionService permissionService)
+    {
+        permissionService.AssertCanCreateOrg();
+        var userId = loggedInContext.User.Id;
+        var orgId = Guid.NewGuid();
+        dbContext.Orgs.Add(new Organization()
+        {
+            Id = orgId,
+            Name = name,
+            Members =
+            [
+                new OrgMember() { Role = OrgRole.Admin, UserId = userId }
+            ]
+        });
+        await dbContext.SaveChangesAsync();
+        return dbContext.Orgs.Where(o => o.Id == orgId);
+    }
+
+    /// <summary>
+    /// set the role of a member in an organization, if the member does not exist it will be created
+    /// </summary>
+    /// <param name="dbContext"></param>
+    /// <param name="permissionService"></param>
+    /// <param name="orgId"></param>
+    /// <param name="role">set to null to remove the member</param>
+    /// <param name="emailOrUsername">either an email or a username for the user whos membership to update</param>
+    [Error<DbError>]
+    [Error<NotFoundException>]
+    [UseMutationConvention]
+    [UseFirstOrDefault]
+    [UseProjection]
+    public async Task<IQueryable<Organization>> SetOrgMemberRole(
+        LexBoxDbContext dbContext,
+        IPermissionService permissionService,
+        Guid orgId,
+        OrgRole? role,
+        string emailOrUsername)
```

Comment:
I would expect you to use the user's Guid here. Email is a bit weird, because it can change.

Reply:
This is conceptually the same as adding a user to a project, which we do using an email or username, otherwise the frontend needs to lookup the user somehow which most users don't have permission to do.